test/pkg/steps/kie_asset_library.go

```diff
@@ -0,0 +1,158 @@
+func registerKieAssetLibrarySteps(ctx *godog.ScenarioContext, data *Data) {
+	ctx.Step("^Project kie-asset-re-marshaller is cloned$", data.projectKieAssetReMarshallerIsCloned)
+	ctx.Step("^Project kie-asset-library is cloned$", data.projectKieAssetLibraryIsCloned)
+	ctx.Step("^Project kie-asset-library is built by maven with configuration:$", data.projectKieAssetLibraryIsBuiltByMavenWithConfiguration)
+	ctx.Step("^Project \"([^\"]*)\" is generated by kie-asset-library$", data.projectIsGeneratedByKieAssetLibrary)
+	ctx.Step("^Project \"([^\"]*)\" in kie-asset-library is built by maven$", data.projectInKieAssetLibraryIsBuiltByMaven)
+	ctx.Step("^Project \"([^\"]*)\" assets are re-marshalled by VS Code$", data.projectAssetsAreRemarshalledByVsCode)
+	ctx.Step(`^Deploy (quarkus|springboot) project "([^"]*)" from kie-asset-library with configuration:$`, data.deployKieAssetLibraryProjectOnOpenshift)
+}
+
+func (data *Data) projectKieAssetLibraryIsBuiltByMavenWithConfiguration(table *godog.Table) error {
+
+	mavenConfig := &mappers.MavenCommandConfig{}
+	if table != nil && len(table.Rows) > 0 {
+		err := mappers.MapMavenCommandConfigTable(table, mavenConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return data.localPathBuiltByMavenWithProfileAndOptions(data.KieAssetLibraryLocation, mavenConfig)
+}
+
+func (data *Data) projectKieAssetReMarshallerIsCloned() error {
+	framework.GetLogger(data.Namespace).Info("Cloning kie-asset-re-marshaller project", "URI", KieAssetReMarshallerGitRepositoryURI, "branch", KieAssetReMarshallerGitRepositoryBranch, "clonedLocation", data.KieAssetReMarshallerLocation)
+
+	cloneOptions := &git.CloneOptions{
+		URL:          KieAssetReMarshallerGitRepositoryURI,
+		SingleBranch: true,
+	}
+
+	var err error
+	reference := KieAssetReMarshallerGitRepositoryBranch
+	if len(reference) == 0 {
+		err = cloneRepository(data.KieAssetReMarshallerLocation, cloneOptions)
+	} else {
+		// Try cloning as branch reference
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(reference)
+		err = cloneRepository(data.KieAssetReMarshallerLocation, cloneOptions)
+		// If branch clone was successful then return, otherwise try other cloning options
+		if err == nil {
+			return nil
+		}
+
+		// If branch cloning failed then try cloning as tag
+		cloneOptions.ReferenceName = plumbing.NewTagReferenceName(reference)
+		err = cloneRepository(data.KieAssetReMarshallerLocation, cloneOptions)
+	}
+	return err
+}
+
+func (data *Data) projectKieAssetLibraryIsCloned() error {
+	framework.GetLogger(data.Namespace).Info("Cloning kie-asset-library project", "URI", KieAssetLibraryGitRepositoryURI, "branch", KieAssetLibraryGitRepositoryBranch, "clonedLocation", data.KieAssetLibraryLocation)
+
+	cloneOptions := &git.CloneOptions{
+		URL:          KieAssetLibraryGitRepositoryURI,
+		SingleBranch: true,
+	}
+
+	var err error
+	reference := KieAssetLibraryGitRepositoryBranch
+	if len(reference) == 0 {
```

Comment:
same here, `reference` will not be empty

Reply:
well spotted , thank you